astropy/modeling/fitting.py

```diff
@@ -1214,18 +1214,21 @@ def _wrap_deriv(params, model, weights, x, y, z=None):
                     output = np.array(
                         [np.ravel(_) for _ in np.array(weights) * fit_deriv]
                     )
-                    if output.shape != fit_deriv.shape:
-                        output = np.array(
-                            [np.ravel(_) for _ in np.atleast_2d(weights).T * fit_deriv]
-                        )
-                    return output
+
                 except ValueError:
                     return np.array(
                         [
                             np.ravel(_)
                             for _ in np.array(weights) * np.moveaxis(fit_deriv, -1, 0)
                         ]
                     ).transpose()
+
+                else:
+                    if output.shape != fit_deriv.shape:
+                        output = np.array(
+                            [np.ravel(_) for _ in np.atleast_2d(weights).T * fit_deriv]
+                        )
+                    return output
```

Comment:
This should be part of the the original try block the `ValueError` in question can still be triggered by this block. Even though the CI does not cover this case.

Reply:
Ok. @Shaheer-Ahmd, sorry to have to move it back. When you do, can you add a comment that both parts can trip raise ValueError and should be in the same block.

Reply:
Okay!